cartographer/mapping/internal/2d/ray_casting.h

```diff
@@ -19,21 +19,16 @@
 
 #include <vector>
 
-#include "cartographer/common/port.h"
-#include "cartographer/mapping/2d/probability_grid.h"
-#include "cartographer/sensor/point_cloud.h"
-#include "cartographer/sensor/range_data.h"
 #include "cartographer/transform/transform.h"
 
 namespace cartographer {
 namespace mapping {
 
-// For each ray in 'range_data', inserts hits and misses into
-// 'probability_grid'. Hits are handled before misses.
-void CastRays(const sensor::RangeData& range_data,
-              const std::vector<uint16>& hit_table,
-              const std::vector<uint16>& miss_table, bool insert_free_space,
-              ProbabilityGrid* probability_grid);
+// Compute all pixels in which some part of the line segment connecting 'begin'
+// and 'end' lies with 'subpixel_scale' precision.
+std::vector<Eigen::Array2i> RayToPixelMask(const Eigen::Array2i& begin,
+                                           const Eigen::Array2i& end,
+                                           int subpixel_scale);
```

Comment:
subpixel_scale is misleading here as the returned result is only pixels, I wouldn't expose it as a parameter unless your new feature requires this

Reply:
begin and end are expected to be scaled by subpixel scale, that should be indicated more clearly. As the parameter allows a tradeoff between speed and accuracy, I don't believe forcing a fixed value for all potential use-cases is a good idea.

Reply:
I'd rename the parameters to "begin_in_subpixels",..., "subpixel_per_pixel" to make it clear. (This was not clear to me during the previous review.) "scaled by" in my eyes has two potential meanings.